byway/src/main/java/com/google/sps/servlets/EntityServlet.java

```diff
@@ -0,0 +1,115 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Trip;
+import com.google.sps.data.UserInfo;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/entity")
+public class EntityServlet extends HttpServlet {
```

Comment:
The `get` and `post` methods of this servlet seem moderately unrelated: `GET` returns a list of all trips, while `POST` creates a single new trip.

There's no requirement that you do everying related to things in a single servlet. Instead, it's good to group things so that `GET` and `POST` are meaningfully related (and you don't have to have both). For example, you could have `/api/createtrip` which only supports `POST`, and `/api/gettrips` which only supports `GET`. Better to have your HTTP methods have meaningful names than try to do everything in as few servlets as possible.

Reply:
done